src/gui/FileBrowser.cpp

```diff
@@ -567,10 +567,17 @@ void FileBrowserTreeWidget::keyPressEvent(QKeyEvent * ke )
 	else if (vertical || horizontal || preview || insert) { stopPreview(); }
 
 	// Try to get the currently selected item as a FileItem
-	auto file = dynamic_cast<FileItem*>(currentItem());
-	// If it's null (folder, separator, etc.), there's nothing left for us to do
-	if (file == nullptr) { return; }
+	QTreeWidgetItem* item = currentItem();
 
+	auto file = dynamic_cast<FileItem*>(item);
+
+	// If it's not a file, try to expand it.
+	if (file == nullptr) {
+		if (preview) {
+			item->setExpanded(!item->isExpanded());
+		}
+		return;
+	}
```

Comment:
Just improving this a bit.
```suggestion
	// If it's not a file and the item is being previewed, assume it's a directory and try to expand or shrink it
	if (!file && preview)
	{
			item->setExpanded(!item->isExpanded());
			return;
	}
```

Reply:
actually @sakertooth I realised why i wrote it this way.

We want it to return if its not a file no matter what. So by checking if its a preview in the same if statement, it will only return if its a preview. This might cause issues.